runners/core-java/src/main/java/org/apache/beam/runners/core/SimpleDoFnRunner.java

```diff
@@ -1190,12 +1190,12 @@ public Timer withOutputTimestamp(Instant outputTimestamp) {
      * </ul>
      */
     private void setAndVerifyOutputTimestamp() {
-
       if (outputTimestamp != null) {
         checkArgument(
-            !outputTimestamp.isBefore(elementInputTimestamp),
-            "output timestamp %s should be after input message timestamp or output timestamp of firing timers %s",
+            !outputTimestamp.plus(fn.getAllowedTimestampSkew()).isBefore(elementInputTimestamp),
```

Comment:
We should probably explicitly handle Long.MAX_VALUE here to make sure it does not overflow. Or is that handled by the Instant itself?

Reply:
Seems like it's not handled by the instant. I copied what is done above which is to just special case the MAX_VALUE. Technically I think this could still overflow if it's almost the MAX_VALUE? But seems like it's pretty unlikely that it would be almost max value but not max value. I also added a test for the max value case.